api/client/service/opts.go

```diff
@@ -235,7 +235,8 @@ func (m *MountOpt) Type() string {
 func (m *MountOpt) String() string {
 	mounts := []string{}
 	for _, mount := range m.values {
-		mounts = append(mounts, fmt.Sprintf("%v", mount))
+		repr := fmt.Sprintf("%s %s %s", mount.Type, mount.Source, mount.Target)
```

Comment:
The issue is that in https://github.com/docker/docker/pull/23581/files#diff-bc77f096de902ba0e4925cc05db09f8eR97, you're testing the output of an inconsequential string type. It's just more code in the way of future changes.

These kinds of things have their way of becoming syntax over time.